stream.go

```diff
@@ -238,18 +238,25 @@ func (s *Stream) sendWindowUpdate() error {
 
 	// Determine the delta update
 	max := s.session.config.MaxStreamWindowSize
-	delta := max - atomic.LoadUint32(&s.recvWindow)
```

Comment:
Why did the atomic load go away? I don't have too much context around this code, but seems like the receive lock `recvLock` has a lot more contention since many other places use it, vs previously here it was using atomic load to get a potentially faster lower level primitive to load `recvWindow`.  Can this new logic for bufLen be rewritten to use atomic.loadUint32 for both `recvWindow` and `recvBuf.len()` so we don't need the recvLock mutex?

Reply:
Hi @preetapan, thanks for the feedback.

We must synchronize access to the `recvWindow` and `recvBuf` values, because they are related. The previous (incorrect) logic only looked at the `recvWindow` for determining the value of the window update after a `Read` operation. The bug surfaced when a `Read` does not completely drain the `recvBuf`. We must then take remaining bytes into consideration when calculating the final window update value and therefore take the lock.